docs: Add documentation for vibe coding

patrikbraborec opened the pull request:
The PR introduces draft of a documentation for vibe coders. Still a few TODOs:

- [ ] Decide whatever to mentioned context7, or not
- [ ] Make sure that rules and instruction can be copied 
- [ ] Ask for feedback in `#product-docs`
- [ ] Improve the structure, and potentially add more info

apify-service-account commented:
Preview for this PR was built for commit [`41880e6`](https://github.com/apify/apify-docs/commit/41880e64c98ef42227e810e8ad88dcc9bfbee180) and is ready at https://pr-1756.preview.docs.apify.com!

apify-service-account commented:
Preview for this PR was built for commit [`9fc8097`](https://github.com/apify/apify-docs/commit/9fc8097fd1c8ef979eb62c01aa17ae9b4ac0e300) and is ready at https://pr-1756.preview.docs.apify.com!

apify-service-account commented:
Preview for this PR was built for commit [`c55b5ee`](https://github.com/apify/apify-docs/commit/c55b5ee00b23d7877e1f8740e1bea591a1e11ccf) and is ready at https://pr-1756.preview.docs.apify.com!

apify-service-account commented:
Preview for this PR was built for commit [`e1d585e`](https://github.com/apify/apify-docs/commit/e1d585ee6d34c346fbecd3b406133e809685076c) and is ready at https://pr-1756.preview.docs.apify.com!

apify-service-account commented:
Preview for this PR was built for commit [`e1d585e`](https://github.com/apify/apify-docs/commit/e1d585ee6d34c346fbecd3b406133e809685076c) and is ready at https://pr-1756.preview.docs.apify.com!

apify-service-account commented:
Preview for this PR was built for commit [`d516b0c`](https://github.com/apify/apify-docs/commit/d516b0c852131ad8f7a98a10fc48a9622e4039bf) and is ready at https://pr-1756.preview.docs.apify.com!

TC-MO commented:
For what it's worth I took a deep dive after this review to double check it, and couldn't find any documentation that would refer to this practice as `Vibe coding` , it's always some variaton of `Build with AI`, `AI optimization`, `Build with LLMs`

Examples:

* [Astro](https://docs.astro.build/en/guides/build-with-ai/)
* [Netlify](https://docs.netlify.com/build/build-with-ai/overview/)
* [Mintlify](https://mintlify.com/docs/guides/assistant)
* [Stripe](https://docs.stripe.com/building-with-llms)
* [Angular](https://angular.dev/ai)
* [ReadMe](https://docs.readme.com/main/docs/owlbot)
* [Auth0](https://auth0.com/docs/get-started/auth0-guide)
* [Svelte](https://svelte.dev/docs/llms)

I would strongly recommend `Build with AI` or some variation thereof and don't mention vibe coding in docs

TC-MO commented on sources/platform/actors/development/quick_start/vibe_coding.md:
```diff
@@ -0,0 +1,104 @@
+
+### Use llms.txt and llms-full.txt
+
+LLMs don't automatically discover llms.txt files, you need to add the link manually. Some tools like [Cursor](https://www.cursor.com/) provide settings for this.
```
```suggestion
LLMs don't automatically discover `llms.txt` files, you need to add the link manually. Some tools like [Cursor](https://www.cursor.com/) provide settings for this.
```

TC-MO commented on sources/platform/actors/development/quick_start/vibe_coding.md:
```diff
@@ -0,0 +1,104 @@
+
+To get the most from AI IDEs, add rules or instructions.
+
+See how to set up rules for your AI IDEs:
```
BTW, this might be minor point but is referring to them as `AI IDE` correct? I guess they all have capability to become an IDE through extensions but refer to themselves as Code editors. Just something to think about, I'm open to discussion

TC-MO commented on sources/platform/actors/development/quick_start/vibe_coding.md:
```diff
@@ -0,0 +1,104 @@
+
+Search engines weren't built for Large Language Models (LLMs), but they needs context. That's why we've created `[llms.txt](https://docs.apify.com/llms.txt)` and `[llms-full.txt](https://docs.apify.com/llms-full.txt)` for our documentation. These files follow the [growing standard](https://llmstxt.org/) for LLMs consumption.
+
+| File | Purpose | Impact on LLM Processing |
+| :--- | :--- | :--- |
+| `/robots.txt` | Tells crawlers where to go. | It doesn't help with content understanding. |
+| `/sitemap.xml` | Lists pages. | It doesn't help with content. LLM systems still need to parse complex HTML and handle extra info, which clutters the context window. |
+| `/llms.txt` | Solves LLMs problems by presenting content optimized for LLM processing. | It overcomes context window limits and removes markup and scripts. |
```
Seems like formatting might get the best of us in this case. Table generally looks better but leaves `/` in separate line. Not sure if we can force it to be in the same line
<img width="740" height="315" alt="image" src="https://github.com/user-attachments/assets/7232e63b-5569-4cc6-af79-58891d1dacb0" />

patrikbraborec replied:
Maybe we can remove `/`. I think it is not important in this context. What do you think?

TC-MO reviewed (commented):
A few small things I noticed

apify-service-account commented:
Preview for this PR was built for commit [`3dad123`](https://github.com/apify/apify-docs/commit/3dad123155668f5334a326ecc0d7a98357df6a17) and is ready at https://pr-1756.preview.docs.apify.com!

TC-MO commented on sources/platform/actors/development/quick_start/vibe_coding.md:
```diff
@@ -0,0 +1,104 @@
+
+## `llms.txt` and `llms-full.txt`
+
+Search engines weren't built for Large Language Models (LLMs), but they needs context. That's why we've created `[llms.txt](https://docs.apify.com/llms.txt)` and `[llms-full.txt](https://docs.apify.com/llms-full.txt)` for our documentation. These files follow the [growing standard](https://llmstxt.org/) for LLMs consumption.
```
Sorry while checking preview I noticed I messed up those links here

```suggestion
Search engines weren't built for Large Language Models (LLMs), but they needs context. That's why we've created [`llms.txt`](https://docs.apify.com/llms.txt) and [`llms-full.txt`](https://docs.apify.com/llms-full.txt) for our documentation. These files follow the [growing standard](https://llmstxt.org/) for LLMs consumption.
```

jancurn commented:
But when looking how to vibe code with Apify, why do I need to get a "lecutre" about robots.txt and sitemap ? It adds confusion rather than clarity

apify-service-account commented:
Preview for this PR was built for commit [`88a3937d`](https://github.com/apify/apify-docs/commit/88a3937df44782a826e8131d37f9b493594f59dc) and is ready at https://pr-1756.preview.docs.apify.com!

patrikbraborec commented:
@TC-MO , @jancurn 

I removed the info about site.xml and robots.txt. I agree, it was extra info. I added explanation of llms.txt and llms-full.txt instead. 

I also added a paragraph about markdown pages (this should be later updated once we have buttons as we discussed on slack).

TC-MO reviewed (changes requested):
2 final notes after that it LGTM

patrikbraborec commented:
Thanks, dona.

TC-MO commented on sources/platform/actors/development/quick_start/index.mdx:
```diff
@@ -57,3 +57,14 @@ This is the fastest way to kick-start your Actor development and try out the Api
     />
 </CardGrid>
+
+## Build with AI
```
Any reason why not H3? I thought it would fit nicely as a separate developer path.

TC-MO replied:
Hmm I guess that is true,  we could maybe ponder if it shouldn't then fall under `Start locally` but as far as I am concerned we can do it at later date. thanks!

TC-MO reviewed (approved):
LGTM, one final comment, but more of a question than anything to change
